plot-lib/plot/private/plot2d/interval.rkt

```diff
@@ -412,30 +419,31 @@
          #:line2-style [line2-style (interval-line2-style)]
          #:alpha [alpha (interval-alpha)]
          #:label [label #f]
+         #:add-ticks? [add-ticks? #t]
+         #:far-ticks? [far-ticks? #f]
          #:bandwidth [bandwidth #f]
          #:invert? [invert? #f])
-  (define fail/pos (make-raise-argument-error 'violin x-center ys y-min y-max))
   (define fail/kw (make-raise-keyword-error 'violin))
   (cond
-    [(and y-min (not (rational? y-min)))  (fail/pos "#f or rational" 2)]
-    [(and y-max (not (rational? y-max)))  (fail/pos "#f or rational" 3)]
-    [(and x-min (not (rational? x-min)))  (fail/kw "#f or rational" '#:x-min x-min)]
-    [(and x-max (not (rational? x-max)))  (fail/kw "#f or rational" '#:x-max x-max)]
     [(and y-min (not (rational? y-min)))  (fail/kw "#f or rational" '#:y-min y-min)]
     [(and y-max (not (rational? y-max)))  (fail/kw "#f or rational" '#:y-max y-max)]
+    [(or (> gap 1) (not (rational? gap)))  (fail/kw "real in [0,1]" '#:gap gap)]
+    [(not (rational? skip))  (fail/kw "rational?" '#:skip skip)]
```

Comment:
The type `Nonnegative-Real` should generate a contract that already takes care of the `positive?` restriction.

Reply:
> The type Nonnegative-Real should generate a contract that already takes care of the positive? restriction

Not always.  I don't think the comment below applies to this file now , but it is safer to add the checks to the entry functions into the plot package, since subtle changes to the code can cause crashes:

https://github.com/racket/plot/blob/6da78b761a6005fdfe771318097513adf38c9564/plot-gui-lib/plot/private/gui/plot2d.rkt#L3